### ISSUE TITLE
Gmail number of unread emails isn't updating

### ISSUE THREAD

**Issue by heffneil** [wavebox/waveboxapp#1177]
* Wavebox Version: 4.11.11
* Operating System & Version: OSX 13.15.2
* Gmail

**Expected & actual behavior**
Gmail is showing on the left bar some number of open emails.  Marking emails read doesn't update the number. Right clicking and reloading also doesn't update the number.  Only way to correct the wrong number is close and re-open the application (Wavebox) entirely.

**Is the bug persistent or intermittent?**
Do you only see it sometimes or does it always happen? Seems to happen consistently.  May only occur when the laptop goes to sleep and comes back awake (early morning for me)

**Comment by Thomas101** [comment]
Have you checked the inbox type in Wavebox is set the same as in Gmail? There's some info here -https://wavebox.io/kb-article/

There are also some special configurations which the KB article covers at the bottom

**Comment by heffneil** [comment]
![Screen Shot 2019-12-30 at 12 28 51 PM](https://user-images.githubusercontent.com/22274732/71592902-feea0c80-2aff-11ea-9eed-af071c3cb306.png)
![Screen Shot 2019-12-30 at 12 28 38 PM](https://user-images.githubusercontent.com/22274732/71592903-feea0c80-2aff-11ea-88f4-24c5889386db.png)
These have been my settings for over a year.  Please advise if I am incorrect?  The KB seems to indicate this is correct.

**Comment by Thomas101** [comment]
Yep that looks correct. Maybe try removing the account and adding it back in again. It could be a glitch on Google's end and just clearing everything out might fix it.

If that doesn't work, can you open up wavebox mini from the tray and locate what the app thinks is unread. Then try clicking through a couple to see if clears them.

You mentioned that you we're using reload, if you right click on the app and pick resync does that fetch the correct count?

**Comment by heffneil** [comment]
removed and re-added gmail.  Its not updating the left bar, or the mini on the top of the window.  Its only showing the 5 messages but I have 7 unread right now.  Resync and reload doesn't fix it.  The only way to fix it is to close wavebox and re-open.  Let me know if I can be of any assistance to debug from here...

**Comment by Thomas101** [comment]
Hey, sorry for the delayed response. You can grab some extra debug info you do the following...

* Open the account Settings & locate the Advanced section under Gmail
* Click the `Log Sync network requests` button

This will start logging all the network requests that the Gmail sync makes and open up the location that the log is located. If you right-click on the account and pick `Resync` you should get a request log there.

Can you send a copy of this? (Feel free to blank out any email addresses or send directly to support@wavebox.io)

**Comment by kcinman11358** [comment]
@heffneil I have the same issue. What I have noticed is if I get an email that belongs to conversation, wavebox isn't counting it. So let say I get an email and a reply to that email. Wavebox will display 1 while Gmail will display 2

![image](https://user-images.githubusercontent.com/7409848/71931811-3e4dd400-3164-11ea-8f15-d51b1ab26a28.png)

**Comment by Thomas101** [comment]
@kcinman11358 does restarting the app temporarily fix the problem for you too? If so, are you able to grab the network logs as per https://github.com/wavebox/waveboxapp/issues/1177#issuecomment-571094857

**Comment by kcinman11358** [comment]
nope. i've had this problem for a while now. I can still give you logs if you need them

**Comment by kcinman11358** [comment]
![image](https://user-images.githubusercontent.com/7409848/72300372-380a9c80-3629-11ea-9466-e7d4c4a8ffb7.png)

**Comment by Thomas101** [comment]
@kcinman11358 do you have conversation view enabled or disabled in Gmail under settings? If you it's disabled, does enabling it temporarily make the counts align a little better?

Can you also send a screenshot of your Gmail inbox settings, just so I can double check the inbox settings Vs what's configured in Wavebox

**Comment by kcinman11358** [comment]
@Thomas101 
I have conversation turned off. It's not practical IMO for business use. People will reply to a thread and then reply to another. The "conversation" gets hard to follow 
![image](https://user-images.githubusercontent.com/7409848/72357381-7eefa500-36b0-11ea-9ceb-736711fd3257.png)

I think this is the area you are looking for a screenshot
![image](https://user-images.githubusercontent.com/7409848/72357459-a34b8180-36b0-11ea-9bba-3bb9451c8769.png)

I have tried setting wavebox to be default inbox type, but then the count doesn't work at all
![image](https://user-images.githubusercontent.com/7409848/72357607-daba2e00-36b0-11ea-9860-4336570a3742.png)

**Comment by heffneil** [comment]
so the new wavebox 10 doesn't appear to have this problem - I don't think.
Still new to me but I will let you know.

Thanks,

Neil H.

**Comment by Thomas101** [comment]
@kcinman11358 I think you need to change your inbox type to be "Unread first" in Wavebox. Check the special cases section in https://wavebox.io/kb/gmail-inbox-type

**Comment by heffneil** [comment]
I was wrong - the new version suffers from the same problem.  gmail show 8
and wavebox is stuck on 5.
Thanks,

Neil H.

**Comment by kcinman11358** [comment]
@Thomas101 unread first still has the same miscount
![image](https://user-images.githubusercontent.com/7409848/72442285-d6097e80-3771-11ea-9c34-3203030d30e7.png)

**Comment by heffneil** [comment]
Restarting Wavebox doesn't even fix the number displayed compared to
gmail.  Currently wavebox shows 5, gmail shows 7.

Thanks,

Neil H.

**Comment by heffneil** [comment]
I don't have time to open a new ticket but the latest version of Wavebox 10
doesn't open links in the default browser even after selecting the option
and restarting the program.

Thanks,

Neil H.

**Comment by Thomas101** [comment]
@heffneil can you send a screenshot of your inbox settings in Gmail and wavebox the same way kcinman did

**Comment by heffneil** [comment]
I sent it before

Thanks,

Neil H.

**Comment by Thomas101** [comment]
Whoops sorry!

I have seen a bug a little while ago where Gmail would have a bunch of old emails stuck as being unread. You should be able to check this in the new Wavebox.

In your Gmail account enter https://mail.google.com/mail/feed/atom/ into the address bar. This should give you the feed of what's unread along with the count etc. Do all the emails that are reported there look reasonably recent or are there any old/irrelevant ones?

You might be able to clear any that are stuck by using the URL in the feed and coping the URL and pasting it into the address bar for each message

E.g. in the feed entry below you'd visit `https://mail.google.com/mail/u/0?account_id=tom&message_id=12fb7af3d71b461&view=conv&extsrc=atom`


```xml
<entry>
<title>Daily Agenda as of 5 am</title>
<summary>
Google Calendar, here is your schedule for: Sat 11 Jan 2020
</summary>
<link rel="alternate" href="https://mail.google.com/mail/u/0?account_id=tom&message_id=12fb7af3d71b461&view=conv&extsrc=atom" type="text/html"/>
<modified>2020-01-11T05:05:38Z</modified>
<issued>2020-01-11T05:05:38Z</issued>
<id>tag:gmail.google.com,2004:1656052200815363745</id>
<author>
<name>Google Calendar</name>
<email>calendar-notification@google.com</email>
</author>
</entry>
```

**Comment by heffneil** [comment]
In the new version? I imported from the previous version. The previous version we confirmed it was correct?

Thanks,
Neil H.

**Comment by Thomas101** [comment]
> In the new version? I imported from the previous version. The previous version we confirmed it was correct?

In version 10. This should show exactly what information Google is returning and is what the app uses. We've seen instances before where old emails (e.g. ones which are 3-4 years old) continue to report in the unread feed. Then by locating and opening them, it seems to clear them

**Comment by heffneil** [comment]
Gmail shows 14 and Wavebox still shows 4.  See images below

[image: Screen Shot 2020-01-20 at 8.08.52 AM.png]
[image: Screen Shot 2020-01-20 at 8.08.41 AM.png]
Thanks,

Neil H.

**Comment by Thomas101** [comment]
@heffneil the screenshots haven't attached from the email. You may need to use GitHub on the web to upload them...

**Comment by heffneil** [comment]
Can we please fix the problem where links in Wavebox X do not open in the browser?  I have that option set and every link still opens within its own browser?

![Screen Shot 2020-01-20 at 8 08 52 AM](https://user-images.githubusercontent.com/22274732/72925752-3889ee00-3d21-11ea-83f5-ef2de238760d.png)
![Screen Shot 2020-01-20 at 8 08 41 AM](https://user-images.githubusercontent.com/22274732/72925766-3e7fcf00-3d21-11ea-8894-69f3984bff5d.png)

**Comment by kcinman11358** [comment]
@heffneil , can we stay on topic of the post?

**Comment by heffneil** [comment]
well are you able to find the problem or do you need more details

**Comment by Thomas101** [comment]
@heffneil Can you drop support a message (https://wavebox.io/support) and we can dig a little deeper into what's gone wrong here
